---
layout: post
title: AIZU ONLINE ジャッジでRuntime Errorが発生した
date: 2018-07-17 10:16:56
categories: python python3
---
<!-- {% raw %} -->
<p><a href="http://judge.u-aizu.ac.jp/onlinejudge/" rel="nofollow noreferrer">AIZU ONLINE JUDGE</a>の0009の問題を解くために以下のようなコードを提出したのですが、「<a href="http://judge.u-aizu.ac.jp/onlinejudge/review.jsp?rid=3036169#1" rel="nofollow noreferrer">Runtime Error</a>」が出てしまいます。簡単な問題かもしれませんが原因がわかる方、お答えください。</p>

<pre><code>import math
import sys

def prime_calc(n):
    rootN = math.floor(math.sqrt(n))
    prime = [2]
    data = [i + 1 for i in range(2,n,2)]
    while True:
        p = data[0]
        if rootN &lt;= p:
            return len(prime + data)
        prime.append(p)
        data = [e for e in data if e % p != 0]

def main():
    l = []

    for line in sys.stdin:
        l.append(int(line))        

    for line in l:
        print(prime_calc(line))

if __name__ == "__main__":
    main()
</code></pre>

<p>追記:tanalab2 さんの解答のとおりにしたのですが、なぜか「<a href="http://judge.u-aizu.ac.jp/onlinejudge/review.jsp?rid=3036328#1" rel="nofollow noreferrer">Runtime Error</a>」が出てしまいます。原因は何でしょうか？</p>

<p>追記:tanalab2さんのご指摘の通りのコードにしました。</p>

<p>追記:以下の入力を試しましたが問題はありませんでした。</p>

<pre><code>10
3
11
</code></pre>
<!-- {% endraw %} -->
